geode-core/src/main/java/org/apache/geode/internal/cache/BucketPersistenceAdvisor.java

```diff
@@ -461,7 +461,18 @@ public void endBucketCreation(PersistentMemberID newId) {
       }
       atomicCreation = false;
     }
-    super.setOnline(false, true, newId);
+    try {
+      superSetOnline(false, true, newId);
+    } catch (ReplyException replyException) {
+      synchronized (lock) {
```

Comment:
It might be best to transform `atomicCreation` into an `AtomicBoolean` to avoid repeated uses of `synchronized`. The inside the previous `synchronized` block only seems to check or set the flag as well as some logging.

Reply:
There is other places in the code path that doing operations like setOnline when holding the lock. Whether we change atomicCreation to AtomicBoolean or not, we still need to hold the lock before modifying the value to avoid multiple threads creating same bucket concurrently.